.golangci.yml

```diff
@@ -56,49 +61,96 @@ linters-settings:
   funlen:
     # Checks the number of lines in a function.
     # If lower than 0, disable the check.
-    # Default: 60
-    lines: 60 
+    lines: 200
     # Checks the number of statements in a function.
-    # Default: 40
-    statements: 40
+    statements: 80
 
+  dupl:
+    # Tokens count to trigger issue.
+    threshold: 200
+
+  nestif:
+    # Minimal complexity of if statements to report.
+    min-complexity: 10
 
 linters:
   enable-all: true
```

Comment:
> forbidigo for print statements in tests, is there a point in going through a logger that in the end still prints?

I think those should at least just be `t.Logf()` and not `fmt.Printf()`. Only exception for `fmt.Printf()` should be `cmd/...` IMO.

I agree with `errname`, we usually use `ErrXXX` indeed.

The other two I don't really have an opinion on.

Reply:
>I think those should at least just be t.Logf() and not fmt.Printf(). Only exception for fmt.Printf() should be cmd/... IMO.

Makes sense

Reply:
Cool I think it makes sense to disable `cyclop` and `errname` since it's a majority opinion here.

As for `nilnil`, I think it's nice to have it as it forces us to rethink our functions to return more safe results. The only downside I can think of is the mocks we currently use. On that front, I think the direction we want to push is to use `mock` library to mock interfaces to better control our test flow. So I don't have a strong opinion here and my thought is to leave it here and try it first. If it turns out to be bad, we can then disable it.

> Probably a lot of changes required? Personally not a fan of screaming capitals, especially Htlc vs HTLC

As for `stylecheck`, this linter is really OCD-friendly😂 imho, it's better to be HTLC since it's an acronym, while ofc you can argue it's a word. With `stylecheck` we can put this arguement in peace and stick to that.

What do you think?

Reply:
Cool `nilnil` is now disabled. I guess we can always review the code and decide then. Meanwhile, although I have a very light opinion over what should be capitalized, I'm more interested in unifying the format - that is, whether it's HTLC or Htlc, as long as we write them in the same format, I'm happy😂. And I think `stylecheck` helps us reach that goal.